pkg/minishift/addon/manager/addon_manager.go

```diff
@@ -266,9 +279,37 @@ func (m *AddOnManager) verifyRequiredVariablesInContext(context *command.Executi
 
 	if len(missingVars) > 0 {
 		missing := strings.TrimSpace(strings.Join(missingVars, ", "))
-		return errors.New(fmt.Sprintf("The variable(s) %s are required by the add-on, but are not defined in the context.", missing))
+		return fmt.Errorf("The variable(s) %s are required by the add-on, but are not defined in the context", missing)
+	}
+
+	return nil
+}
+
+func (m *AddOnManager) verifyRequiredOpenshiftVersion(context *command.ExecutionContext, meta addon.AddOnMeta) error {
+	dockerCommander := context.GetDockerCommander()
+	versionInfo, err := dockerCommander.Exec(" ", constants.OpenshiftContainerName, "openshift", "version")
+	if err != nil {
+		return err
 	}
 
+	// verionInfo variable have below string as value along with new line
```

Comment:
ok, works for me. I still argue a regexp using a pattern buffer would be easier: 'openshift v([0-9]+\.[0-9]+\.[0-9]+\.)+.*'

And then you get the version from the first pattern buffer. 
But if you want to go ahead with string splitting, that's fine with me

Reply:
@hferentschik I mostly find regex bit hard to understand (my personal opinion) :)